dipy/tracking/local/tests/test_tissue_classifier.py

```diff
@@ -0,0 +1,173 @@
+
+    # test voxel center
+    for ind in np.ndindex(mask.shape):
+        pts = np.array(ind, dtype='float64')
+        state_boolean = btc_boolean.check_point(pts)
+        state_float64 = btc_float64.check_point(pts)
+        if mask[ind] > 0:
+            npt.assert_equal(state_boolean, 1)
+            npt.assert_equal(state_float64, 1)
+        else:
+            npt.assert_equal(state_boolean, 2)
+            npt.assert_equal(state_float64, 2)
+
+    # test random points in voxel
+    for ind in np.ndindex(mask.shape):
+        for _ in range(10):
+            pts = np.array(ind, dtype='float64') + np.random.random(3) - 0.5
+            state_boolean = btc_boolean.check_point(pts)
+            state_float64 = btc_float64.check_point(pts)
+            if mask[ind] > 0:
+                npt.assert_equal(state_boolean, 1)
+                npt.assert_equal(state_float64, 1)
+            else:
+                npt.assert_equal(state_boolean, 2)
+                npt.assert_equal(state_float64, 2)
+
+    # test outside points
+    outside_pts = [[100, 100, 100], [0, -1, 1], [0, 10, 2]]
+    for pts in outside_pts:
+        pts = np.array(pts, dtype='float64')
+        state_boolean = btc_boolean.check_point(pts)
+        state_float64 = btc_float64.check_point(pts)
+        npt.assert_equal(state_boolean, -1)
+        npt.assert_equal(state_float64, -1)
+
+
+def test_threshold_tissue_classifier():
+    """This tests that the thresholdy tissue classifier return expected
+     tissue types:
+
+        OUTSIDEIMAGE = -1
+        INVALIDPOINT = 0
+        TRACKPOINT = 1
+        ENDPOINT = 2
+    """
+    # enum TissueClass (tissue_classifier.pxd) is not accessible
+    # from here. To be changed when minimal cyhton version > 0.21.
+    # cython 0.21 - cpdef enum to export values into Python-level namespace
+    # https://github.com/cython/cython/commit/50133b5a91eea348eddaaad22a606a7fa1c7c457
+
+    tissue_map = np.random.random((4, 4, 4))
+
+    ttc = ThresholdTissueClassifier(tissue_map.astype('float32'), 0.5)
+
+    # test voxel center
+    for ind in np.ndindex(tissue_map.shape):
+        pts = np.array(ind, dtype='float64')
+        state = ttc.check_point(pts)
+        if tissue_map[ind] > 0.5:
+            npt.assert_equal(state, 1)
+        else:
+            npt.assert_equal(state, 2)
+
+    # test random points in voxel
+    inds = [[0, 1.4, 2.2], [0, 2.3, 2.3], [0, 2.2, 1.3], [0, 0.9, 2.2],
+            [0, 2.8, 1.1], [0, 1.1, 3.3], [0, 2.1, 1.9], [0, 3.1, 3.1]]
+    for pts in inds:
+        pts = np.array(pts, dtype='float64')
+        state = ttc.check_point(pts)
+        res = scipy.ndimage.map_coordinates(
+            tissue_map, np.reshape(pts, (3, 1)), order=1, mode='nearest')
+        if res > 0.5:
+            npt.assert_equal(state, 1)
+        else:
+            npt.assert_equal(state, 2)
+
+    # test outside points
+    outside_pts = [[100, 100, 100], [0, -1, 1], [0, 10, 2]]
+    for pts in outside_pts:
```

Comment:
As before I need more points very close to the boundaries. This is the critical area.